zerver/lib/reminders.py

```diff
@@ -0,0 +1,66 @@
+from django.conf import settings
+from django.utils.translation import gettext as _
+
+from zerver.actions.message_send import internal_send_private_message
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.markdown.fenced_code import get_unused_fence
+from zerver.lib.mention import silent_mention_syntax_for_user
+from zerver.lib.message import access_message
+from zerver.lib.message_cache import MessageDict
+from zerver.lib.url_encoding import near_message_url, topic_narrow_url
+from zerver.models import Message, Stream, UserProfile
+from zerver.models.users import get_system_bot
+
+
+def get_reminder_formatted_content(
+    message: Message, current_user: UserProfile, stream: Stream | None
+) -> str:
+    if message.is_stream_message():
+        assert stream is not None
+        narrow_link = topic_narrow_url(
+            realm=current_user.realm,
+            stream=stream,
+            topic_name=message.topic_name(),
+        )
```

Comment:
This should be a `near` link including the message ID, I think?

Reply:
Yeah, I am using a `near` link for the `said` text below, this link is for the topic URL.

Reply:
I kinda feel like you want to use it for both, or at least a `/with/` link.

Reply:
Let's discuss this in the #design thread, we can tweak in a follow-up as we finalize the end-to-end experience.

Reply:
Okay, I will just open a thread when I open a PR for this to implement the frontend part. Using `/with/` link here makes sense.